control-plane-agent/api/proxy/proxy.go

```diff
@@ -0,0 +1,287 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2024 Intel Corporation
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+package proxy
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"sync"
+
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+
+	pb "control-plane-agent/api/proxy/proto/mediaproxy"
+	"control-plane-agent/internal/event"
+	"control-plane-agent/internal/model"
+	"control-plane-agent/internal/registry"
+	"control-plane-agent/internal/telemetry"
+)
+
+type Config struct {
+	ListenPort int
+}
+
+type API struct {
+	pb.UnimplementedProxyAPIServer
+
+	cfg Config
+	srv *grpc.Server
+
+	requestMap   map[string]model.CommandRequest
+	requestMapMx sync.Mutex
+}
+
+func NewAPI(cfg Config) *API {
+	return &API{
+		cfg:        cfg,
+		requestMap: make(map[string]model.CommandRequest),
+	}
+}
+
+func (a *API) CancelCommandRequest(reqId string) {
+	a.requestMapMx.Lock()
+	req, ok := a.requestMap[reqId]
+	a.requestMapMx.Unlock()
+	if ok {
+		delete(a.requestMap, reqId)
```

Comment:
Don't we have to hold the lock during deletion too?

Reply:
Oh, nice catch! Will fix it.

Reply:
Fixed